### PR TITLE
Rel cascaded replica

### PR DESCRIPTION
**Pull request by vibhorkumar123**

### CODE REVIEW

File: roles/setup_efm/tasks/efm_configure.yml [hunk 1]
```diff
@@ -37,6 +37,12 @@
 - name: Prepare final parameters list
   set_fact:
     efm_node_parameters: "{{ efm_default_parameters + efm_node_parameters }}"
+  when: item.value.upstream_node is not defined
```
Comment (jt-edb) [inline roles/setup_efm/tasks/efm_configure.yml:40]: Please use `upstream_node_private_ip`. We use this key to sort the server list: https://github.com/EnterpriseDB/edb-ansible/blob/REL-2_0_3/filter_plugins/sort_servers.py#L90

File: roles/setup_efm/tasks/efm_configure.yml [hunk 2]
```diff
@@ -37,6 +37,12 @@
...
+
+- name: Prepare final parameters list
+  set_fact:
+    efm_node_parameters: "{{ efm_default_parameters + efm_node_parameters + efm_cascaded_replica_param }}"
+  when: item.value.upstream_node is defined
```
Comment (jt-edb) [inline roles/setup_efm/tasks/efm_configure.yml:45]: Please use `upstream_node_private_ip`.

File: roles/setup_replication/tasks/main.yml [hunk 3]
```diff
@@ -46,9 +46,11 @@
     - item.node_type == 'primary'
   loop: "{{ server_list }}"
 
-- name: Prepare slot names array
+- name: Prepare slot names array for primary
   include_tasks: prepare_standby_slots_list.yml
-  when: item.node_type == 'standby'
+  when:
+    - item.node_type == 'standby'
+    - item.upstream_node is not defined
```
Comment (jt-edb) [inline roles/setup_replication/tasks/main.yml:53]: Please use `upstream_node_private_ip`.

File: roles/setup_replication/tasks/pg_basebackup.yml [hunk 4]
```diff
@@ -17,15 +17,33 @@
     pg_basebackup: "{{ pg_basebackup + ' --username=' + pg_replication_user }}"
   when: pg_replication_user|length > 0
 
+- name: Set host and port
+  set_fact:
+    pg_basebackup: "{{ pg_basebackup + ' --host=' + item.upstream_node + ' --port=' + pg_port }}"
```
Comment (jt-edb) [inline roles/setup_replication/tasks/pg_basebackup.yml:22]: Please use `upstream_node_private_ip`.

File: roles/setup_replication/tasks/pg_basebackup.yml [hunk 5]
```diff
@@ -17,15 +17,33 @@
...
+  when:
+    - not use_hostname
+    - item.upstream_node is defined
+
+- name: Set host and port
+  set_fact:
+    pg_basebackup: "{{ pg_basebackup + ' --host=' + upstream_hostname + ' --port=' + pg_port }}"
+  when:
+    - use_hostname
+    - item.upstream_node is defined
```
Comment (jt-edb) [inline roles/setup_replication/tasks/pg_basebackup.yml:32]: Please use `upstream_node_private_ip`.

File: roles/setup_replication/tasks/pg_basebackup.yml [hunk 6]
```diff
@@ -17,15 +17,33 @@
...
+
 - name: Set host and port
   set_fact:
     pg_basebackup: "{{ pg_basebackup + ' --host=' + primary_private_ip + ' --port=' + pg_port }}"
-  when: not use_hostname
+  when:
+    - not use_hostname
+    - item.upstream_node is not defined
```
Comment (jt-edb) [inline roles/setup_replication/tasks/pg_basebackup.yml:39]: Please use `upstream_node_private_ip`.

File: roles/setup_replication/tasks/pg_basebackup.yml [hunk 7]
```diff
@@ -17,15 +17,33 @@
...
 
 - name: Set host and port
   set_fact:
     pg_basebackup: "{{ pg_basebackup + ' --host=' + primary_host_name + ' --port=' + pg_port }}"
-  when: use_hostname
+  when:
+    - use_hostname
+    - item.upstream_node is not defined
```
Comment (jt-edb) [inline roles/setup_replication/tasks/pg_basebackup.yml:46]: Please use `upstream_node_private_ip`.

File: roles/setup_replication/tasks/setup_replication.yml [hunk 8]
```diff
@@ -31,6 +23,12 @@
   when: use_hostname
   delegate_to: "{{ item.public_ip }}"
 
+- name: Call upstream update based on the upstream node
+  block:
+    - import_tasks: upstream_node.yml
+  when:
+    - item.upstream_node is defined
```
Comment (jt-edb) [inline roles/setup_replication/tasks/setup_replication.yml:30]: Please use `upstream_node_private_ip`.

File: roles/setup_replication/tasks/upstream_node.yml [hunk 9]
```diff
@@ -0,0 +1,29 @@
+---
+
+- name: Get upstream node public ip
+  set_fact:
+    upstream_public_ip: "{{ server.public_ip }}"
+  when:
+    - item.upstream_node is defined
+    - server.private_ip == item.upstream_node
```
Comment (jt-edb) [inline roles/setup_replication/tasks/upstream_node.yml:8]: Please use `upstream_node_private_ip`.

File: roles/setup_replication/tasks/upstream_node.yml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+  loop_control:
+    loop_var: server
+
+
+- name: Get upstream node hostnames
+  set_fact:
+    upstream_hostname: "{{ server.hostname }}"
+  when:
+    - item.upstream_node is defined
+    - server.private_ip == item.upstream_node
```
Comment (jt-edb) [inline roles/setup_replication/tasks/upstream_node.yml:20]: Please use `upstream_node_private_ip`.

Comment (vibhorkumar123) [reply roles/setup_replication/tasks/upstream_node.yml]: Fixed! Changed upstream_node to upstream_node_private_ip

File: roles/setup_efm/tasks/efm_configure.yml (hunk 1 above)
Comment (vibhorkumar123) [reply roles/setup_efm/tasks/efm_configure.yml]: Fixed

**Review by jt-edb — APPROVED** [review]

**Pull request merged**